### PR TITLE
fix(gatsby): Fix static query parsing for for a special case

### LINKED ISSUE #21212: GraphQL problem parsing query in gatsby@2.18.20 and later
After upgrading from `gatsby@2.18.19-jobs-api-v2.39` to `gatsby@2.18.20` (and all later versions) I get this error

>  ERROR #85915  GRAPHQL
There was a problem parsing the GraphQL query in file:
`src/components/Nav/index.js`

The [file in question](https://github.com/janosh/janosh.io/blob/master/src/components/Nav/index.js) looks pretty harmless. So maybe the changes introduced a bug? The [log states](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/CHANGELOG.md#21820-2020-01-09):

> **Bug Fixes**
gatsby: Webpack /_ webpackPrefetch/webpackPreload _/ comments causing a bug (#20403) (1b89b10)
**Features**
gatsby: Allow alternative import syntax for useStaticQuery (#20330) (17eaa72)

#20330 sounds like it might be related.

### Environment

<details>
<summary>`gatsby info --clipboard`</summary>

  System:
    OS: macOS 10.15.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.13.1 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.6 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 79.0.3945.130
    Safari: 13.0.5
  npmPackages:
    gatsby: 2.18.20 => 2.18.20 
    gatsby-image: 2.2.40 => 2.2.40 
    gatsby-plugin-algolia: ^0.5.0 => 0.5.0 
    gatsby-plugin-catch-links: 2.1.25 => 2.1.25 
    gatsby-plugin-favicon: ^3.1.6 => 3.1.6 
    gatsby-plugin-feed-mdx: ^1.0.1 => 1.0.1 
    gatsby-plugin-google-analytics: 2.1.35 => 2.1.35 
    gatsby-plugin-lodash: ^3.1.16 => 3.1.20 
    gatsby-plugin-manifest: 2.2.41 => 2.2.41 
    gatsby-plugin-mdx: ^1.0.59 => 1.0.70 
    gatsby-plugin-netlify-cache: ^1.2.0 => 1.2.0 
    gatsby-plugin-offline: 3.0.34 => 3.0.34 
    gatsby-plugin-react-helmet: 3.1.22 => 3.1.22 
    gatsby-plugin-sharp: 2.4.5 => 2.4.5 
    gatsby-plugin-styled-components: ^3.1.19 => 3.1.19 
    gatsby-remark-autolink-headers: ^2.1.19 => 2.1.24 
    gatsby-remark-code-titles: ^1.1.0 => 1.1.0 
    gatsby-remark-copy-linked-files: ^2.1.31 => 2.1.37 
    gatsby-remark-embed-video: 2.0.1 => 2.0.1 
    gatsby-remark-emojis: ^0.4.2 => 0.4.2 
    gatsby-remark-images: 3.1.44 => 3.1.44 
    gatsby-remark-katex: ^3.1.18 => 3.1.24 
    gatsby-remark-responsive-iframe: 2.2.32 => 2.2.32 
    gatsby-remark-smartypants: 2.1.21 => 2.1.21 
    gatsby-remark-sub-sup: ^1.0.0 => 1.0.0 
    gatsby-remark-vscode: ^2.0.1 => 2.0.1 
    gatsby-source-filesystem: 2.1.48 => 2.1.48 
    gatsby-transformer-sharp: 2.3.14 => 2.3.14 
    gatsby-transformer-yaml: 2.2.24 => 2.2.24 
</details>

### PR DESCRIPTION
**Pull request by vladar**
## Description

Before this PR following component would have caused an error during static query parsing:

```js
import { graphql, useStaticQuery } from 'gatsby'

export { Bar } from 'bar'

export default () => {
  const data = useStaticQuery(graphql`query StaticQueryName { foo }`);
  return <div>{data.doo}</div>;
}
```

This was happening because of the presence of `export { Bar } from 'bar'` which was treated incorrectly while trying to follow static query variable chain

This issue was introduced in #20330

## Related Issues

Fixes #21212

### CODE REVIEW

File: packages/gatsby/src/query/file-parser.js [hunk 1]
```diff
@@ -366,6 +366,9 @@ async function findGraphQLTags(
         // Look for exported page queries
         traverse(ast, {
           ExportNamedDeclaration(path, state) {
+            if (path.node.source) {
```
Comment (pvdz) [inline packages/gatsby/src/query/file-parser.js]: Technically the property has to be `null` if there was no `from`. I guess in practice it doesn't matter much.

**Comment by vladar** [comment]
When a component has a StaticQuery we scan all of its exports and follow those exported variables to determine if they lead to this static query (so that we could return it).

But in this edge case of re-exporting the code was failing (basically because there was no scope for this query).

I will add a comment explaining this. I actually tried first but it was confusing as hell. Somehow explaining the same thing on GitHub is easier %)

**Comment by jfrolich** [comment]
Thanks! I obviously didn't think about this type of export in the previous PR 🙈, sorry about that.

**Comment by vladar** [comment]
@jfrolich It is really hard to cover all cases, unfortunately. So no worries at all and thanks for the contribution!

**Comment by vladar** [comment]
Published in `gatsby@2.19.19`

**Pull request merged**